src/Orbit.js

```diff
@@ -239,12 +240,25 @@ class Orbit {
   }
 
   getFile(hash, callback) {
-    // request('http://localhost:8080/ipfs/' + hash, function (error, response, body) {
-    //   if(!error && response.statusCode === 200) {
-        if(callback) callback(null);
-        // if(callback) callback(body);
-    //   }
-    // })
+    if (callback) {
+      callback(null);
+      return;
+    }
+    return this.ipfs.files.cat(hash).then(res => {
+      return new Promise((resolve, reject) => {
+        const buffers = [];
+        res
+          .on('error', err => {
+            reject(err);
+          })
+          .on('data', data => {
+            buffers.push(data);
+          })
+          .on('end', () => {
+            resolve(new Blob(buffers));
+          });
+      });
+    });
   }
```

Comment:
Here's the general problem with catting a file: we can't buffer it into the memory before we pass it back to the requester. This prevents eg. video streaming, which means immediate playback. If we have 2GB video file, it'll take a long time before the video plays (not to mention mem constraints) and this is not the desirable user experience.

That being said, this is part of the architecture refactoring I mentioned I want to do. My plan here is to provide the following function:
```javascript
streamFile(hash) {
  return this.ipfs.files.cat(hash); // returns a Promise that resolves to a Stream
}
```

This is (was?) tricky to implement because the mismatching IPFS APIs and the fact that ipfs.files.cat didn't return a Promise. It might've changed in the past 3 weeks but I haven't had the change to confirm that it is.